ibis/expr/api.py

```diff
@@ -2683,6 +2683,21 @@ def _table_materialize(table):
         return TableExpr(op)
 
 
+def add_column(table, expr, name=None):
```

Comment:
I think we should deprecate this functionality after 1 or 2 more release cycles. It's pretty redundant.

Reply:
That seems like a useful tool. I'm okay with introducing that as a dependency.